src/main/java/net/codejava/MvcController.java

```diff
@@ -193,6 +198,11 @@ public String submitWithdraw(@ModelAttribute("user") User user) {
     String userPassword = jdbcTemplate.queryForObject(getUserPasswordSql, String.class);
 
     if (userPasswordAttempt.equals(userPassword)) {
+      // Make sure withdraw value is positive
+      if (user.getAmountToDeposit() <= 0 || user.getAmountToDeposit() > user.getBalance()) {

```

Comment:
You should use `user.getAmountToWithdraw()` here, not `user.getAmountToDeposit()`. Also, I appreciate you including the check that the withdraw amount should not be greater than the available balance, but that is beyond the scope of what you were tasked to do.